pilot/pkg/xds/debug.go

```diff
@@ -302,7 +302,11 @@ func (s *DiscoveryServer) Syncz(w http.ResponseWriter, req *http.Request) {
 // Can be combined with the push debug interface to reproduce changes.
 func (s *DiscoveryServer) registryz(w http.ResponseWriter, req *http.Request) {
 	all := s.Env.ServiceDiscovery.Services()
-	writeJSON(w, all, req)
+	copied := make([]*model.Service, len(all))
+	for _, svc := range all {
+		copied = append(copied, svc.DeepCopy())
```

Comment:
maybe it has been fixed by #39058, whatever we need a test

Reply:
I don't this it is fixed by that. The concurrent map write might still come if JSON encoder is encoding AddressMap while it is modified. There is already a test but it is difficult to create a test for this I guess. Any ideas?

Reply:
@hzxuzhonghu any thoughts?

Reply:
I believe there is no mutating of `AddressMap` before deepcopy now. So this can not happen now.

Reply:
Ok. Cool I will close this